ios/RNDeviceInfo/RNDeviceInfo.m

```diff
@@ -314,4 +306,22 @@ - (NSDictionary *)constantsToExport
     resolve(@(batteryLevel));
 }
 
+RCT_EXPORT_METHOD(getUserAgent:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        if (_webView == nil) {
+            _webView = [[WKWebView alloc] initWithFrame:CGRectZero];
+        }
+        [_webView evaluateJavaScript:@"navigator.userAgent" completionHandler:^(id result, NSError * error) {
+            if (error == nil) {
+                if (result != nil) {
```

Comment:
This looks like promise will never be resolved with if `result == nil`.
Can we do better like:

```
if (result) {
  resolve([NSString stringWithFormat:@"%@", result]); // still not sure if this is necessary but I guess docs don't guarantee type of the result...
} else {
  resolve(nil)
}
```
makes sense?

Reply:
I saw `resolve(nil)` in many libs, but that's not too critical :)